core/src/main/java/com/graphhopper/reader/osm/WaySegmentParser.java

```diff
@@ -224,14 +221,27 @@ public void handleNode(ReaderNode node) {
 
             acceptedNodes++;
 
-            // we keep node tags for barrier nodes
+            // remember which nodes we want to split
             if (splitNodeFilter.test(node)) {
                 if (nodeType == JUNCTION_NODE) {
                     LOGGER.debug("OSM node {} at {},{} is a barrier node at a junction. The barrier will be ignored",
                             node.getId(), Helper.round(node.getLat(), 7), Helper.round(node.getLon(), 7));
                     ignoredSplitNodes++;
                 } else
+                    nodeData.setSplitNode(node.getId());
+            }
+
+            // store node tags if at least one important tag is included and make this available for the edge handler
+            for (Map.Entry<String, Object> e : node.getTags().entrySet()) {
+                if (INCLUDE_IF_NODE_TAGS.contains(e.getKey())) {
+                    node.removeTag("created_by");
+                    node.removeTag("source");
+                    node.removeTag("note");
+                    node.removeTag("fixme");
+                    node.setTags(new HashMap<>(node.getTags())); // create compact Map
```

Comment:
I'd rather create the compact map in OSMNodeData.setTags, so it is kind of up to this 'storage' class how to the tags are stored (and compacted and such).